api/handle_scenario_testrun.go

```diff
@@ -37,3 +38,38 @@ func handleCreateScenarioTestRun(uc usecases.Usecases) func(c *gin.Context) {
 		})
 	}
 }
+
+func handleListScenarioTestRun(uc usecases.Usecases) func(c *gin.Context) {
+	return func(c *gin.Context) {
+		ctx := c.Request.Context()
+		organizationId, err := utils.OrganizationIdFromRequest(c.Request)
+		if presentError(ctx, c, err) {
+			return
+		}
+		scenarioID := c.Query("scenario_id")
+
+		usecase := usecasesWithCreds(ctx, uc).NewScenarioTestRunUseCase()
+		testruns, err := usecase.ListTestRunByScenarioId(ctx, organizationId, scenarioID)
```

Comment:
We have a little pain point in most of the backend, that I have not had time to resolve yet, which is that we use "string" fields in models & dtos to contain UUIDs. 
Since the usecase here clearly expects a (non empty) value for scenarioID, let's please validate its presence and the format in the handler (especially here because scenario_id comes from the query, so could be seen as optional)

Reply:
in this case the format is already specified in the oas specification.. which will perform a first step of validation.